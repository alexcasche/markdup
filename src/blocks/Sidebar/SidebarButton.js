import { applyStyleModifiers } from 'styled-components-modifiers'
import { shade } from 'polished'
import { Button } from 'elements'

const MODIFIERS = {
  new: ({ theme }) => `
    margin-top: 15px;
    background-color: ${shade(0.85, theme.colors.blue)};
    &:hover:not(:active) {
      background-color: ${theme.colors.blue};
    }
  `,
  save: ({ theme }) => `
    background-color: ${shade(0.85, theme.colors.green)};
    &:hover:not(:active) {
      background-color: ${theme.colors.green};
    }
  `,
  delete: ({ theme }) => `
    background-color: transparent;
    font-size: 14px;
    border: 1px solid ${shade(0.75, theme.colors.grey)};
    color: ${shade(0.75, theme.colors.grey)};
    &:hover:not(:active) {
      border: 1px solid ${theme.colors.grey};
      color: ${theme.colors.grey};
    }
  `,
}

const SidebarButton = Button.extend`
  margin: 10px auto;
  width: calc(100% - 30px);
  height: 44px;
  font-size: 16px;
  color: ${props => props.theme.colors.black};
  ${applyStyleModifiers(MODIFIERS)};
`

export default SidebarButton
