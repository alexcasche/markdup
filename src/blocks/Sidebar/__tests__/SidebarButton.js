import SidebarButton from '../SidebarButton'
import { shade } from 'polished'

describe('SidebarButton Element', () => {
  it('renders correctly', () => {
    const tree = render(<SidebarButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds new modifier', () => {
    const tree = render(<SidebarButton modifiers={['new']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule(
      'background-color',
      shade(0.85, theme.colors.blue)
    )
  })

  it('adds save modifier', () => {
    const tree = render(<SidebarButton modifiers={['save']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule(
      'background-color',
      shade(0.85, theme.colors.green)
    )
  })

  it('adds delete modifier', () => {
    const tree = render(<SidebarButton modifiers={['delete']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule(
      'border',
      `1px solid ${shade(0.75, theme.colors.grey)}`
    )
  })
})
