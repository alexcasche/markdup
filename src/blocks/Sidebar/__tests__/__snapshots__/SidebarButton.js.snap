// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarButton Element adds delete modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  margin: 10px auto;
  width: calc(100% - 30px);
  height: 44px;
  font-size: 16px;
  color: #1D1F21;
  background-color: transparent;
  font-size: 14px;
  border: 1px solid #808186;
  color: #808186;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover:not(:active) {
  border: 1px solid #ABADB3;
  color: #ABADB3;
}

<button
  className="c0"
/>
`;

exports[`SidebarButton Element adds new modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  margin: 10px auto;
  width: calc(100% - 30px);
  height: 44px;
  font-size: 16px;
  color: #1D1F21;
  margin-top: 15px;
  background-color: #31aed1;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover:not(:active) {
  background-color: #3ACDF6;
}

<button
  className="c0"
/>
`;

exports[`SidebarButton Element adds save modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  margin: 10px auto;
  width: calc(100% - 30px);
  height: 44px;
  font-size: 16px;
  color: #1D1F21;
  background-color: #2cd35d;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover:not(:active) {
  background-color: #34F96E;
}

<button
  className="c0"
/>
`;

exports[`SidebarButton Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  margin: 10px auto;
  width: calc(100% - 30px);
  height: 44px;
  font-size: 16px;
  color: #1D1F21;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
/>
`;
